docs/proposals/compute-resource-metrics-api.md

```diff
@@ -115,13 +116,11 @@ only be available through the kubelet. The types of metrics are detailed
 Additionally, all endpoints (except root discovery endpoint) support the
 following optional query parameters:
 
-- `start` - start time to return metrics from; type json encoded
-  `time.Time`; since samples are retrieved at discrete intervals, the first
-  sample after the start time is the actual beginning.
-- `end` - end time to return metrics to; type json encoded `time.Time`
-- `step` - the time step between each stats sample; type int (seconds), default
```

Comment:
I think that we should start from the most recent sample/end of the query window. Starting from the fixed time 0 would be hard to understand, especially if step is not a divisor of 60. 

The other possibility is return the newest sample from  the step-sized window, starting from end. In your example it would be:
start=100 stop=201 step=30  => [200, 170, 140, 110]
 or if there were some "holes" [197, 169, 135, 105]
This will prevent going completely out of sync in case some missing metrics/hiccups.